### PR TITLE
Compatible with Vungle Android SDK 6.3.17

### PR DESCRIPTION
**Pull request by ashinagawa**

### CODE REVIEW

**Comment by ashinagawa** [comment]
Thanks for your review @chauduyphanvu1 
1. Yes. You are correct. We removed requirement for all Placement IDs (which a user had to set in "Custom Event Class Data" or MoPub platform sends them to adapter as part of  `seerverExtra`). But, we still require "individual placement ID" to link an AdUnit to a Vungle's placement. 

2. Can you please try UnitIDs below?
"06afd52c99834e2f95b31084aff8e958”- Interstitial
"77815b57f8e041a6affde816b4b47f4b"- Interstitial
"e38b724f4f6041ac828d6dc87b85ef6f"- Interstitial
"3ab5b17178f84b5caba811528fc9c5c2" - Rewarded
"e26cf45489a040fba1219eb2feca7b76" - Rewarded
"e804a88f93304bf68a308625240c6ef1" - Rewarded

3. I will verify this issue asap. Please note that "FlexFeed" ad won't work with adapter since SDK will return view (where we render the flex feed ad) and there is no api between MoPub and Vungle adapter to handle it at a moment.

**Comment by ashinagawa** [comment]
@chauduyphanvu1  

I verified #3 ("E/Vungle: Failed to download assets...") internally, and this is just trowing an error but it should not crash. I couldn't reproduce the crash with this error. If you still crash on your end, please send me log.

Also, we updated our SDK, so please use 
`implementation 'com.github.vungle:vungle-android-sdk:6.3.17'`

File: Vungle/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
 ## Changelog
+  * 6.3.17.0
+    * This version of the adapters has been certified with Vungle 6.13.17.
+    * Deprecate the use of placement IDs at Vungle's initialization time per Vungle's new SDK.
```
Comment (rsatrasala) [inline Vungle/CHANGELOG.md]: How about "Remove the placement IDS (pids) parameter from Vungle's init call per Vungle's new SDK requirements"

File: Vungle/com/mopub/mobileads/VungleInterstitial.java [hunk 2]
```diff
@@ -83,7 +82,7 @@ public void run() {
         }
 
         if (!sVungleRouter.isVungleInitialized()) {
-            sVungleRouter.initVungle(context, mAppId, mPlacementIds);
+            sVungleRouter.initVungle(context, mAppId);
```
Comment (rsatrasala) [inline Vungle/com/mopub/mobileads/VungleInterstitial.java]: We should add a comment here as "Remove the placement IDS (pids) parameter from `initVungle` method per Vungle's SDK 6.3.17 changes""

File: Vungle/com/mopub/mobileads/VungleInterstitial.java [hunk 3]
```diff
@@ -148,27 +147,9 @@ private boolean validateIdsInServerExtras(Map<String, String> serverExtras) {
...
-            for (String pid : mPlacementIds) {
-                if (pid.equals(mPlacementId)) {
-                    foundInList = true;
-                }
-            }
-            if (!foundInList) {
-                MoPubLog.w(INTERSTITIAL_TAG + "Placement IDs for this Ad Unit is not in the array of Placement IDs");
-                isAllDataValid = false;
-            }
+            MoPubLog.w(INTERSTITIAL_TAG + "No need to set placement IDs " +
```
Comment (rsatrasala) [inline Vungle/com/mopub/mobileads/VungleInterstitial.java]: publishers don't set placement IDs field. This field is auto generated by our ad server. So we can remove this block

File: Vungle/com/mopub/mobileads/VungleRewardedVideo.java [hunk 4]
```diff
@@ -67,20 +65,19 @@ protected String getAdNetworkId() {
 
     @Override
     protected boolean checkAndInitializeSdk(@NonNull final Activity launcherActivity,
-                                            @NonNull final Map<String, Object> localExtras,
-                                            @NonNull final Map<String, String> serverExtras) throws Exception {
+            @NonNull final Map<String, Object> localExtras,
```
Comment (rsatrasala) [inline Vungle/com/mopub/mobileads/VungleRewardedVideo.java]: The format seems to be changing. We should keep tab or space consistent across adapters

File: Vungle/com/mopub/mobileads/VungleRewardedVideo.java [hunk 5]
```diff
@@ -67,20 +65,19 @@ protected String getAdNetworkId() {
...
             }
 
             if (!validateIdsInServerExtras(serverExtras)) {
                 mAppId = VUNGLE_DEFAULT_APP_ID;
-                mPlacementIds = VUNGLE_DEFAULT_ALL_PLACEMENT_IDS;
             }
 
             if (!sVungleRouter.isVungleInitialized()) {
-                sVungleRouter.initVungle(launcherActivity, mAppId, mPlacementIds);
+                sVungleRouter.initVungle(launcherActivity, mAppId);
```
Comment (rsatrasala) [inline Vungle/com/mopub/mobileads/VungleRewardedVideo.java]: Comment about init method change here as well

File: Vungle/com/mopub/mobileads/VungleRewardedVideo.java [hunk 6]
```diff
@@ -169,27 +166,9 @@ private boolean validateIdsInServerExtras (Map<String, String> serverExtras) {
...
-            for (String pid:  mPlacementIds) {
-                if(pid.equals(mPlacementId)) {
-                    foundInList = true;
-                }
-            }
-            if(!foundInList) {
-                MoPubLog.w(REWARDED_TAG + "Placement IDs for this Ad Unit is not in the array of Placement IDs");
-                isAllDataValid = false;
-            }
+            MoPubLog.w(REWARDED_TAG + "No need to set placement IDs " +
```
Comment (rsatrasala) [inline Vungle/com/mopub/mobileads/VungleRewardedVideo.java]: We can remove this block as explained above

File: Vungle/com/mopub/mobileads/VungleRouter.java [hunk 7]
```diff
@@ -82,7 +80,7 @@ public void onSuccess() {
 
                 // Pass the user consent from the MoPub SDK to Vungle as per GDPR
                 boolean canCollectPersonalInfo = MoPub.canCollectPersonalInformation();
-                Vungle.updateConsentStatus(canCollectPersonalInfo ? Vungle.Consent.OPTED_IN : Vungle.Consent.OPTED_OUT);
+                Vungle.updateConsentStatus(canCollectPersonalInfo ? Vungle.Consent.OPTED_IN : Vungle.Consent.OPTED_OUT, "");
```
Comment (rsatrasala) [inline Vungle/com/mopub/mobileads/VungleRouter.java]: @ashinagawa Please add a one liner comment about the new parameter of updateConsentStatus

File: Vungle/com/mopub/mobileads/VungleRouter.java [hunk 8]
```diff
@@ -156,13 +154,13 @@ public void playAdForPlacement(String placementId, AdConfig adConfig) {
     }
 
     public void updateConsentStatus(Vungle.Consent status) {
-        Vungle.updateConsentStatus(status);
+        Vungle.updateConsentStatus(status, "");
```
Comment (rsatrasala) [inline Vungle/com/mopub/mobileads/VungleRouter.java]: @ashinagawa request the same comment as above here

**Pull request merged**